src/Setup.ts

```diff
@@ -2,35 +2,64 @@
 // https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runspre
 
 import * as core from '@actions/core';
+import { HttpClient } from '@actions/http-client';
 import * as tc from '@actions/tool-cache';
 import * as path from 'path';
 import uuidV4 from 'uuid/v4';
 
 export const sqlcmdToolName = 'go-sqlcmd';
-export const sqlcmdVersion = '1.0.0';
+export const sqlcmdDownloadUrl = 'https://github.com/microsoft/go-sqlcmd/releases/latest';
+export const sqlcmdFallbackVersion = '1.6.0';
 
 export default class Setup {
     
     /**
      * Ensures go-sqlcmd is in the runner's tool cache and PATH environment variable.
      */
     public static async setupSqlcmd(): Promise<void> {
+
+        // Determine latest version or use fallback version
+        let sqlcmdVersion = await this.extractVersionFromLatestRelease(sqlcmdDownloadUrl);
+        if (sqlcmdVersion) {
+            core.debug(`Using latest go-sqlcmd version: ${sqlcmdVersion}`);
+        } else {
+            sqlcmdVersion = sqlcmdFallbackVersion;
+            core.debug(`Failed to get latest go-sqlcmd version. Using fallback version: ${sqlcmdVersion}`);
+        }
+
         // Get sqlcmd from tool cache; if not found, download it and add to tool cache
         let sqlcmdPath = tc.find(sqlcmdToolName, sqlcmdVersion);
         if (!sqlcmdPath) {
-            const extractedPath = await this.downloadAndExtractSqlcmd();
+            const extractedPath = await this.downloadAndExtractSqlcmd(sqlcmdVersion);
             sqlcmdPath = await tc.cacheDir(extractedPath, sqlcmdToolName, sqlcmdVersion);
         }
         
         // Add sqlcmd to PATH
         core.addPath(sqlcmdPath);
     }
 
+    /**
+     * Gets the version number from the latest release based on the redirected URL.
+     * @returns The version number or undefined if the version could not be parsed or HTTP request failed.
+     */
+    public static async extractVersionFromLatestRelease(releaseUrl: string): Promise<string | undefined> {
+        const http = new HttpClient('Azure/sql-action', undefined, { allowRedirects: false });
```

Comment:
Do we have access to user token from github action? Since this is unauthenticated, it will be much more rate-limited.

If we do, may be easier just to use octokit library instead of httpclient

Reply:
Unfortunately it doesn't appear there's an easy way to get the token, we'd have to introduce a new input so users can pass in `GITHUB_TOKEN` to.

Reply:
I'm not sure what the rate limiting will be here - since we're not using a REST API necessarily, we're checking the redirect.